cpp/source/deal.II/mp_values.h

```diff
@@ -41,107 +41,107 @@ class MPValues
 
   MPValues() = default;
 
-  MPValues(MPValuesParameters<dim, spacedim> const &params);
-
   virtual ~MPValues() = default;
 
   virtual void get_values(std::string const &key,
                           active_cell_iterator const &cell,
-                          std::vector<double> &values) const;
+                          std::vector<double> &values) const = 0;
+};
+
+template <int dim, int spacedim = dim>
```

Comment:
why do you have spacedim? The geometry doesn't support spacedim!=dim. And I don't see why you would want to do MPValues<2,3>?